src/hooks/UseForm/useForm/useForm.ts

```diff
@@ -0,0 +1,458 @@
+import React, { useCallback, useMemo, useRef } from 'react'
+import { CoordinatesOfLength, CoordinatesOrNever, Graph } from '../../UseGraph/Graph'
+import useGraph from '../../UseGraph/useGraph'
+import {
+  Changed,
+  Errors,
+  FieldElement,
+  Fields,
+  FieldsData,
+  FormData,
+  HandleSubmit,
+  HandleSubmitOptions,
+  PartialDataKeys,
+  Register,
+  RegisterOptions,
+  RegisterResult,
+  Touched,
+  UseForm,
+  UseFormOptions,
+  UseFormReturn,
+} from './types'
+import {
+  getElementDefaultValue,
+  getOnChangeValue,
+  getTypedData,
+  getTypedFieldValue,
+  isBlank,
+} from './utils'
+
+/**
+ * Helps manage form state.
+ * Provides a `register` function for registering fields with the hook and handles tracking
+ * their values, errors, and submission of the form.
+ *
+ * @template {FieldsData} TData
+ * @returns {{ register: Register<TData>; errors: Errors<TData>; touched: Touched<TData>; handleSubmit: HandleSubmit<TData>; }}
+ */
+const useForm: UseForm = <TData extends FieldsData, TDimensions extends number = 0>(
+  {
+    dimensions,
+    isRequiredErrorMessageOverride,
+    // @ts-expect-error TDimensions defaults to 0 if dimensions is not defined, so this is okay
+  }: UseFormOptions<TDimensions> = { dimensions: 0 },
+): UseFormReturn<TData, TDimensions> => {
+  const fieldsGraph = useRef<FormData<Fields<TData, TDimensions>, TDimensions>>(
+    new Graph<Fields<TData, TDimensions>, TDimensions>({ dimensions }),
+  )
```

Comment:
Doing so causes lint errors about `fieldsGraph` not being included in dependency arrays, which it should not be. I think it is preferable to just access it with `.current` where used rather than complicating the dependency arrays.